pkg/pipeline/utils/prom_server.go

```diff
@@ -22,17 +22,26 @@ import (
 	"os"
 
 	"github.com/netobserv/flowlogs-pipeline/pkg/api"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/sirupsen/logrus"
 )
 
+var handlerRegistered bool = false
+
 // StartPromServer listens for prometheus resource usage requests
-func StartPromServer(tlsConfig *api.PromTLSConf, server *http.Server, panicOnError bool) {
+func StartPromServer(tlsConfig *api.TLSConf, server *http.Server, panicOnError bool, reg *prometheus.Registry) {
 	logrus.Debugf("entering StartPromServer")
 
 	// The Handler function provides a default handler to expose metrics
 	// via an HTTP server. "/metrics" is the usual endpoint for that.
-	http.Handle("/metrics", promhttp.Handler())
+	h := promhttp.HandlerFor(reg, promhttp.HandlerOpts{})
+	server.Handler = h
+	// TODO: This needs more work. enable different endpoints for different handlers, and register them on distinct ServeMux
+	if !handlerRegistered {
+		http.Handle("/metrics", promhttp.Handler())
+		handlerRegistered = true
+	}
```

Comment:
Hey @KalmanMeth ,
I think this should work to address your comment:

```suggestion
	mux := http.NewServeMux()
	mux.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
	server.Handler = mux
```